File: Server/src/main/java/pt/ist/ccu2021/queued/Server/dto/CompanyAccountDto.java
package pt.ist.ccu2021.queued.Server.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.*;
import pt.ist.ccu2021.queued.Server.domain.CompanyAccount;
import pt.ist.ccu2021.queued.Server.domain.UserAccount;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString
@JsonPropertyOrder({"id", "name", "email", "password", "staffEmail", "staffPassword"})
public class CompanyAccountDto {

    @JsonProperty("id")
    private int id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("email")
    private String email;

    @JsonProperty("password")
    private String password;

    @JsonProperty("staffEmail")
    private String staffEmail;

    @JsonProperty("staffPassword")
    private String staffPassword;

    public CompanyAccountDto(CompanyAccount company){
        id = company.getId();
        name = company.getName();
        email = company.getEmail();
        password = company.getPassword();
        staffEmail = company.getStaffEmail();
        staffPassword = company.getStaffPassword();
    }
    public CompanyAccount toDomain(){
        return CompanyAccount.builder().name(name).email(email)
                .password(password).staffEmail(staffEmail).staffPassword(staffPassword).build();
    }
}

File: staff/lib/dto/CompanyAccountDto.dart
class CompanyAccountDto {

  CompanyAccountDto({this.id, this.name, this.email, this.password, this.staffEmail, this.staffPassword});

  int id;

  String name;

  String email;

  String password;

  String staffEmail;

  String staffPassword;

  CompanyAccountDto.fromJson(Map<String, dynamic> json) {
    this.id = json['id'];
    this.name = json['name'];
    this.email = json['email'];
    this.password = json['password'];
    this.staffEmail = json['staffEmail'];
    this.staffPassword = json['staffPassword'];
  }

  Map<String, dynamic> toJson(){
      return {'id': this.id.toString(), 'name': this.name, 'email': this.email, 'password': this.password, 'staffEmail': this.staffEmail, 'staffPassword': this.staffPassword};
  }


}
